Packs/PcapAnalysis/Scripts/PcapFileExtractor/PcapFileExtractor.py

```diff
@@ -0,0 +1,195 @@
+from CommonServerUserPython import *  # noqa: E402 lgtm [py/polluting-import]
+import os
+from os.path import isfile
+from tempfile import mkdtemp
+import shutil
+import glob
+import subprocess
+import hashlib
+
+
+def get_pcap_path(args):
+    """
+    :param args: arg from demisto
+    :return: path of pcap file
+    """
+    file_entry_id = ''
+    if 'entry_id' in args:
+        file_entry_id = args.get('entry_id')  # type: ignore
+
+    if not file_entry_id:
+        return_error('You must set entry_id')
+
+    res = demisto.executeCommand('getFilePath', {'id': file_entry_id})
+    if res[0]['Type'] == entryTypes['error']:
+        return_error(f'Failed to get the file path for entry: {file_entry_id}')
+
+    return res[0]['Contents']['path']
+
+
+def find_files_protocol(file_path):
+    """
+    :param file_path: the pcap file path
+    :return: protocol: the protocol to use while extracting the files
+    """
+
+    protocol = ''
+    data_list = []
+    command_ = f'tshark -r {file_path}'
+    process = subprocess.Popen(command_, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    stdout = process.communicate()
+    stdout = str(stdout).split('\\n')
+    for line in stdout:
+        if line.endswith(')') and 'HTTP' in line:
+            data_list.append(line)
+            protocol = 'http'
+
+        if line.endswith(')') and 'IMF' in line:
+            data_list.append(line)
+            protocol = 'imf'
+
+        if line.endswith(')') and 'SMB' in line:
+            data_list.append(line)
+            protocol = 'smb'
+
+        if line.endswith(')') and 'DICOM' in line:
+            data_list.append(line)
+            protocol = 'dicom'
+
+        if line.endswith(')') and 'TFTP' in line:
+            data_list.append(line)
+            protocol = 'tftp'
+
+    return protocol, data_list
+
+
+def extract_files(file_path, dir_path):
+    """
+    :param file_path: the pcap file path.
+    :param dir_path: directory for the files will be extract to
+    :return:
+        excluded_files: the excludedfiles which are in dir_path
+    """
+    protocol, _ = find_files_protocol(file_path)
+    if not protocol:
+        return_error('Could not find a valid protocol for extracting the files')
+
+    if protocol == 'http':
+        command_ = f'tshark -r {file_path} --export-objects http,{dir_path}'
+
+    elif protocol == 'smb':
+        command_ = f'tshark -r {file_path} --export-objects smb,{dir_path}'
+
+    elif protocol == 'imf':
+        command_ = f'tshark -r {file_path} --export-objects imf,{dir_path}'
+
+    elif protocol == 'tftp':
+        command_ = f'tshark -r {file_path} --export-objects tftp,{dir_path}'
+
+    elif protocol == 'dicom':
+        command_ = f'tshark -r {file_path} --export-objects dicom,{dir_path}'
+
+    process = subprocess.Popen(command_, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    process.communicate()
+    excluded_files = [f for f in glob.glob(dir_path + '/*')]
```

Comment:
why is this needed?

Reply:
This is the list of all files exported from the pcap. we iterate over them in the upload command.